packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -417,6 +420,41 @@ const verifyRules = (parsedRules: RulesParameters, proposal: SnapshotProposalGra
   return { verified: true };
 };
 
+// Check if the proposal has been executed before.
+const hasBeenExecuted = async (
+  currentProposal: TransactionsProposedEvent,
+  params: MonitoringParams
+): Promise<boolean> => {
+  // Get all previous proposals with matching transactions and explanation for the same module that were proposed before
+  // the current proposal.
+  const og = await getOgByAddress(params, currentProposal.address);
+  const previousMatchingProposals = (
+    await runQueryFilter<TransactionsProposedEvent>(og, og.filters.TransactionsProposed(), {
+      start: 0,
+      end: currentProposal.blockNumber,
+    })
+  ).filter(
+    (previousProposal) =>
+      previousProposal.args.proposalHash === currentProposal.args.proposalHash &&
+      previousProposal.args.explanation === currentProposal.args.explanation &&
+      (previousProposal.blockNumber !== currentProposal.blockNumber ||
+        previousProposal.transactionIndex < currentProposal.transactionIndex ||
+        (previousProposal.transactionHash === currentProposal.transactionHash &&
+          previousProposal.logIndex < currentProposal.logIndex))
```

Comment:
Is all of this logic just meant to _not_ catch the current proposal since it would be captured in the same log query? Once again, I think this is what's happening, just want to double check my understanding.

Reply:
Yes, and also to exclude any subsequent proposals in the same tx or block. But I now realize this is redundant as its not possible to execute in the same block due to liveness requirements